packages/main/src/plugin/extension-loader.ts

```diff
@@ -1047,11 +1086,26 @@ export class ExtensionLoader {
     this.extensionState.set(extension.id, 'stopping');
     this.apiSender.send('extension-stopping');
 
+    if (!system) {
+      // extension was explicitly disabled, save the configuration
+      let disabledExtensions = this.getDisabledExtensionIds();
```

Comment:
could help if the method return always array (including empty array) and not undefined